src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -1030,11 +1030,14 @@ FixedwingPositionControl::handle_setpoint_type(const position_setpoint_s &pos_sp
 		}
 
 		if (pos_sp_curr.type == position_setpoint_s::SETPOINT_TYPE_POSITION) {
-			// POSITION: achieve position setpoint altitude via loiter
-			// close to waypoint, but altitude error greater than twice acceptance
+			// Achieve position setpoint altitude via loiter when laterally close to WP.
+			// Detect if system has switchted into a Loiter before (check _position_sp_type), and in that
+			// case remove the dist_xy check (not switch out of Loiter until altitude is reached).
 			if ((!_vehicle_status.in_transition_mode) && (dist >= 0.f)
 			    && (dist_z > _param_nav_fw_alt_rad.get())
-			    && (dist_xy < 2.f * math::max(acc_rad, loiter_radius_abs))) {
+			    && (dist_xy < math::max(acc_rad, loiter_radius_abs)
```

Comment:
@sfuhrer not sure if this conversation is continuing somewhere - but why should this be the maximum considering `loiter_radius_abs`? what if the loiter radius is 200m, but the L1 distance is only 20m (e.g. when the vehicle flies very slow in wind) -- we dont really want it accepted until L1 (or npfg) says we're within the region that we start turning into the path.

Reply:
We partly proceeded with the discussion in https://github.com/PX4/PX4-Autopilot/pull/20992#issuecomment-1413712321.

We certainly can improve it further, removing the 2 factor here was only the very lowest hanging fruit I immediately saw.